aix/SPECS/4.2.0/wazuh-agent-4.2.0-aix.spec

```diff
@@ -101,23 +101,38 @@ if [ $1 = 1 ]; then
 fi
 
 if [ $1 = 2 ]; then
-  if %{_localstatedir}/bin/wazuh-control status 2>/dev/null | grep "is running" > /dev/null 2>&1; then
+  if /etc/rc.d/init.d/wazuh-agent status 2>/dev/null | grep "is running" > /dev/null 2>&1; then
     /etc/rc.d/init.d/wazuh-agent stop > /dev/null 2>&1 || :
     touch %{_localstatedir}/tmp/wazuh.restart
   fi
   %{_localstatedir}/bin/ossec-control stop > /dev/null 2>&1 || %{_localstatedir}/bin/wazuh-control stop > /dev/null 2>&1
 fi
 
-%post
 if [ $1 = 2 ]; then
   if [ -d %{_localstatedir}/logs/ossec ]; then
-    rm -rf %{_localstatedir}/logs/wazuh
-    cp -rp %{_localstatedir}/logs/ossec %{_localstatedir}/logs/wazuh
+    cp -rp %{_localstatedir}/logs/ossec %{_localstatedir}/tmp/logs/wazuh > /dev/null 2>&1
+    rm -rf %{_localstatedir}/logs/ossec/*
+    rm -rf %{_localstatedir}/logs/ossec/.??*
   fi
 
   if [ -d %{_localstatedir}/queue/ossec ]; then
+    cp -rp %{_localstatedir}/queue/ossec %{_localstatedir}/tmp/queue/sockets > /dev/null 2>&1
+    rm -rf %{_localstatedir}/queue/ossec/*
+    rm -rf %{_localstatedir}/queue/ossec/.??*
+  fi
+fi
+
+%post
+
+if [ $1 = 2 ]; then
+  if [ -d %{_localstatedir}/tmp/logs/wazuh ]; then
+    rm -rf %{_localstatedir}/logs/wazuh
+    mv %{_localstatedir}/tmp/logs/ossec %{_localstatedir}/logs/wazuh> /dev/null 2>&1
+  fi
```

Comment:
I don't understand this block of code, could you explain a bit please?

Reply:
To avoid the problem with the `%posttrans` section we separated the copy of `logs/ossec` and `queue/ossec` into their new location in 2 steps one  in the `%pre `section and another in the `%post`. 

For the `cp`, we ignore the output to avoid showing the error caused by copying a socket, and after that, we remove the contents of the directories so RPM can remove them, without showing a warning.